tests/integration/ie-tests.sh

```diff
@@ -324,264 +433,24 @@ cleanup_user_failure() {
-      --conf spark.hadoop.fs.s3a.connection.ssl.enabled=false \
-      --conf spark.hadoop.fs.s3a.path.style.access=true \
-      --conf spark.hadoop.fs.s3a.endpoint=$S3_ENDPOINT \
-      --conf spark.hadoop.fs.s3a.access.key=$ACCESS_KEY \
-      --conf spark.hadoop.fs.s3a.secret.key=$SECRET_KEY 
-    ')" > pyspark.out
-
-  cat pyspark.out
-  l=$(cat pyspark.out | grep "Number of lines" | tail -n 1 | rev | cut -d' ' -f1 | rev | cut -c 1-3)
-  echo -e "Number of lines: \n ${l}"
-  rm pyspark.out
-  validate_file_length $l
-}
-
-test_pyspark_s3_in_pod() {
-  run_pyspark_s3_in_pod tests spark
-}
-
-
-test_pyspark_in_pod() {
-  run_pyspark_in_pod tests spark
-}
-
-test_spark_sql_in_pod() {
-  run_spark_sql_in_pod tests spark
-}
-
-test_restricted_account_in_pod() {
-
-  kubectl config set-context spark-context --namespace=tests --cluster=prod --user=spark
-
-  run_example_job_in_pod tests spark
-}
-
-cleanup_user_failure_in_pod() {
-  teardown_test_pod
-  cleanup_user_failure
-}
 
 setup_tests
 
-(setup_user_admin_context && test_example_job && cleanup_user_success) || cleanup_user_failure
+(setup_user_admin_context && test_spark_pi_example && cleanup_user_success) || cleanup_user_failure
 
 (setup_user_admin_context && test_spark_shell && cleanup_user_success) || cleanup_user_failure
 
-(setup_user_admin_context && test_pyspark && cleanup_user_success) || cleanup_user_failure
+(setup_user_admin_context && test_spark_sql_with_s3 && cleanup_user_success) || cleanup_user_failure
 
-(setup_user_admin_context && test_spark_sql && cleanup_user_success) || cleanup_user_failure
+(setup_user_admin_context && test_spark_sql_with_azure_abfss && cleanup_user_success) || cleanup_user_failure
 
-(setup_user_admin_context && test_pyspark_s3 && cleanup_user_success) || cleanup_user_failure
+(setup_user_admin_context && test_pyspark_with_s3 && cleanup_user_success) || cleanup_user_failure
 
-(setup_user_restricted_context && test_restricted_account && cleanup_user_success) || cleanup_user_failure
-
-setup_test_pod
-
-(setup_user_admin_context && test_example_job_in_pod && cleanup_user_success) || cleanup_user_failure_in_pod
+(setup_user_admin_context && test_pyspark_with_azure_abfss && cleanup_user_success) || cleanup_user_failure
 
-(setup_user_admin_context && test_spark_shell_in_pod && cleanup_user_success) || cleanup_user_failure_in_pod
+(setup_user_admin_context && test_example_job_with_s3 && cleanup_user_success) || cleanup_user_failure
 
-(setup_user_admin_context && test_pyspark_in_pod && cleanup_user_success) || cleanup_user_failure_in_pod
+(setup_user_admin_context && test_example_job_with_azure_abfss && cleanup_user_success) || cleanup_user_failure
 
-(setup_user_admin_context && test_spark_sql_in_pod && cleanup_user_success) || cleanup_user_failure_in_pod
-
-(setup_user_admin_context && test_pyspark_s3_in_pod && cleanup_user_success) || cleanup_user_failure_in_pod
-
-teardown_test_pod
```

Comment:
**question** why did we remove the teardown?

Reply:
The pod tests have been removed here (since they are almost duplicates of the tests done in charmed-spark-rock) and since we don't test on pod, the pod deletion and creation logic has been removed as well.